akka-docs/src/main/paradox/typed/extending.md

```diff
@@ -0,0 +1,60 @@
+# Extending Akka
+
+Akka extensions can be used for almost anything, they provide a way to create
+an instance of a class only once for the whole ActorSystem and be able to access
+it from anywhere. Akka features such as Cluster, Serialization and Sharding are all
+Akka extensions. Below is the use-case of managing an expensive database connection 
+pool and accessing it from various places in your application.
+
+You can choose to have your Extension loaded on-demand or at `ActorSystem` creation 
+time through the Akka configuration.
+Details on how to make that happens are below, in the "Loading from Configuration" section.
+
+@@@ warning
+
+Since an extension is a way to hook into Akka itself, the implementor of the extension needs to
+ensure the thread safety of his/her extension.
+
+@@@
+
+## Building an extension
+
+Let's build an extension to manage a shared database connection pool.
+
+Scala
+:  @@snip [ExtensionDocSpec.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/extensions/ExtensionDocSpec.scala) { #shared-resource }
+
+Java
+:  @@snip [ExtensionDocTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/extensions/ExtensionDocTest.java) { #shared-resource }
+
+First create an @apidoc[akka.actor.typed.Extension] , this will be created only once per ActorSystem:
+
+Scala
+:  @@snip [ExtensionDocSpec.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/extensions/ExtensionDocSpec.scala) { #extension }
+
+Java
+:  @@snip [ExtensionDocTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/extensions/ExtensionDocTest.java) { #extension }
+
+This is the public API of your extension. Internally in this example we instantiate our expensive databsae connection.
```

Comment:
Shall we also add something about why it is helpful to do this with an extension over other options? Per system I mean?